apps/emqx_bridge_pulsar/test/emqx_bridge_pulsar_impl_producer_SUITE.erl

```diff
@@ -1054,31 +1068,44 @@ t_resource_manager_crash_before_producers_started(Config) ->
     ),
     ok.
 
-t_cluster(Config) ->
-    MQTTTopic = ?config(mqtt_topic, Config),
-    ResourceId = resource_id(Config),
-    Cluster = cluster(Config),
-    ClientId = emqx_guid:to_hexstr(emqx_guid:gen()),
-    QoS = 0,
-    Payload = emqx_guid:to_hexstr(emqx_guid:gen()),
+t_cluster(Config0) ->
+    ct:timetrap({seconds, 120}),
+    ?retrying(Config0, 3, fun do_t_cluster/1).
+
+do_t_cluster(Config) ->
     ?check_trace(
         begin
+            MQTTTopic = ?config(mqtt_topic, Config),
+            ResourceId = resource_id(Config),
+            Cluster = cluster(Config),
+            ClientId = emqx_guid:to_hexstr(emqx_guid:gen()),
+            QoS = 0,
+            Payload = emqx_guid:to_hexstr(emqx_guid:gen()),
+            NumNodes = length(Cluster),
+            {ok, SRef0} = snabbkaffe:subscribe(
+                ?match_event(#{?snk_kind := emqx_bridge_app_started}),
+                NumNodes,
+                25_000
+            ),
             Nodes = [N1, N2 | _] = start_cluster(Cluster),
             %% wait until bridge app supervisor is up; by that point,
             %% `emqx_config_handler:add_handler' has been called and the node should be
             %% ready to create bridges.
-            NumNodes = length(Nodes),
-            {ok, _} = snabbkaffe:block_until(
-                ?match_n_events(NumNodes, #{?snk_kind := emqx_bridge_app_started}),
-                15_000
-            ),
-            {ok, SRef0} = snabbkaffe:subscribe(
+            {ok, _} = snabbkaffe:receive_events(SRef0),
+            {ok, SRef1} = snabbkaffe:subscribe(
                 ?match_event(#{?snk_kind := pulsar_producer_bridge_started}),
                 NumNodes,
-                15_000
+                25_000
             ),
             {ok, _} = erpc:call(N1, fun() -> create_bridge(Config) end),
-            {ok, _} = snabbkaffe:receive_events(SRef0),
+            {ok, _} = snabbkaffe:receive_events(SRef1),
+            {ok, _} = snabbkaffe:block_until(
+                ?match_n_events(
+                    NumNodes,
+                    #{?snk_kind := bridge_post_config_update_done}
+                ),
+                25_000
+            ),
```

Comment:
We want _both nodes_ to be updated successfully before proceeding.

Reply:
True. I had an impression that's what `emqx_cluster_rpc` does.
https://github.com/emqx/emqx/blob/0211bcf0305ab5f5b53747aae1b3d01636d527f8/apps/emqx_conf/src/emqx_cluster_rpc.erl#L129-L130

Might obviously be some timeouts here or there, but `create_bridge` returned `{ok, _}` at this point, so that seems unlikely.

Reply:
Speaking of the devil...

https://github.com/emqx/emqx/actions/runs/5191063167/jobs/9358606173?pr=10919#step:7:1414